src/with-component.js

```diff
@@ -3,7 +3,10 @@
 import { withChildren } from './with-children';
 import { withProps } from './with-props';
 import { withRenderer } from './with-renderer';
+import { withState } from './with-state';
 import { withUnique } from './with-unique';
 
-export const withComponent = (Base?: Class<any>) =>
-  withChildren(withProps(withRenderer(withUnique(Base || HTMLElement))));
+export const withComponent = (Base?: Class<HTMLElement>) =>
+  withChildren(
```

Comment:
This is growing... and growing... and growing. Perf?

Reply:
I wouldn't think so, why would it?

Reply:
I’m just wary of long prototype chains. Maybe it’s unfounded, though?

Reply:
i'm assuming some inlining magic business or something.. worth testing though! :D

Reply:
I'm happy to just keep it like it is. If there's a problem, we can probably do something around flattening the chain since they don't rely on `super`. I talked to some at length about this at Polymer Summit and, at least Chrome, optimises this.

Reply:
Yeah, most elements already have a long prototype chain anyways, I don't think a half dozen extras will move the needle.